rest_tools/server/arghandler.py

```diff
@@ -0,0 +1,153 @@
+"""Handle argument parsing, defaulting, and casting."""
+
+
+import distutils.util
+import json
+from typing import Any, cast, Dict, List, Optional
+
+import tornado.web
+
+# local imports
+from rest_tools.client.json_util import json_decode
+
+
+class _NoDefaultValue:  # pylint: disable=R0903
+    """Signal no default value, AKA argument is required."""
+
+
+NO_DEFAULT = _NoDefaultValue()
+
+
+def _get_json_body(request_handler: tornado.web.RequestHandler) -> Dict[str, Any]:
+    json_body = json_decode(request_handler.request.body)  # type: ignore[no-untyped-call]
+    return cast(Dict[str, Any], json_body)
```

Comment:
Careful here. I think `[]` is a legal JSON but may cause trouble if you're expecting an object.
Do you have a test that covers someone throwing in naked values with being wrapped in an object?

Reply:
Ah, that's true.  lists are legal json, it doesn't have to be a dict.  I've used this before, and seen it other places.

Reply:
Can you elaborate on this?

Reply:
I believe what @blinkdog means is that `json_body` can be either a dict or a list.  both are valid return types.

Reply:
oh! the type-hint is wrong